File: examples/csharp/CommonCode.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.IO;
using System.Linq;
using System.Net;
using System.Security.Cryptography;
using System.Web;

namespace VendorOpenAPIWebApp
{
    public class CommonCode
    {
        public string GetAppKeySettings(string sKey)
        {
            return ConfigurationManager.AppSettings[sKey].ToString();
        }

        public string GetIPAddress()
        {
            string strHostName = "";
            var strIPAddress = "";
            try
            {
                strHostName = System.Net.Dns.GetHostName();
                IPAddress[] ipaddress = System.Net.Dns.GetHostAddresses(strHostName);
                foreach (IPAddress ip in ipaddress)
                {
                    strIPAddress = ip.ToString();
                }
            }
            catch (Exception ex)
            {
                strIPAddress = "";
            }

            return strIPAddress;
        }

        public class LoginRequestMobileRes
        {
            public ResHeader head = new ResHeader();
            public LoginRequestMobileResBody body = new LoginRequestMobileResBody();
        }

        public class LoginRequestMobileResBody
        {
            public string EmailId { get; set; }
            public string Message { get; set; }
            public int Status { get; set; }
            public string TTManagerId { get; set; }
            public string UserName { get; set; }
            public DateTime ValidUpto { get; set; }
            public string VendorName { get; set; }
        }

        public class ResHeader
        {
            public string responseCode { get; set; }
            public string status { get; set; }
            public string statusDescription { get; set; }
        }
        public class BOCommonReq
        {
            public ReqHeader head = new ReqHeader();
            public BOCommonReqBody body = new BOCommonReqBody();
        }

        public class BOCommonReqBody
        {
            public string ClientCode { get; set; }
            public string FromDate { get; set; }
            public string ToDate { get; set; }
        }

        public class OrderStatusReq
        {
            public ReqHeader head = new ReqHeader();
            public OrderStatusReqBody body = new OrderStatusReqBody();
        }

        public class OrderStatusReqBody
        {
            public string ClientCode { get; set; }
            public List<OrderStatusList> OrdStatusReqList = new List<OrderStatusList>();
        }

        public class OrderStatusList
        {
            public char Exch { get; set; }
            public char ExchType { get; set; }
            public long ScripCode { get; set; }
            public string RemoteOrderID { get; set; }
        }

        public class TradeInformationReq
        {
            public ReqHeader head = new ReqHeader();
            public TradeInformationReqBody body = new TradeInformationReqBody();
        }

        public class TradeInformationReqBody
        {
            public string ClientCode { get; set; }
            public List<TradeInformationList> TradeInformationList = new List<TradeInformationList>();
        }

        public class TradeInformationList
        {
            public char Exch { get; set; }
            public char ExchType { get; set; }
            public long ScripCode { get; set; }
            public string ExchOrderID { get; set; }
        }

        public class PreOrdMarginCalReq
        {
            public ReqHeader head = new ReqHeader();
            public PreOrdMarginCalReqBody body = new PreOrdMarginCalReqBody();
        }

        public class PreOrdMarginCalReqBody
        {
            public string ClientCode { get; set; }
            public string OrderRequestorCode { get; set; }
            public string Exch { get; set; }
            public string ExchType { get; set; }
            public int ScripCode { get; set; }
            public string PlaceModifyCancel { get; set; }
            public string TransactionType { get; set; }
            public string AtMarket { get; set; }
            public double LimitRate { get; set; }
            public string WithSL { get; set; }
            public double SLTriggerRate { get; set; }
            public char IsSLTriggered { get; set; }
            public long Volume { get; set; }
            public long OldTradedQty { get; set; }
            public char ProductType { get; set; }
            public string ExchOrderId { get; set; }
            public string ClientIP { get; set; }
            public int AppSource { get; set; }
        }

        public class CommonReq
        {
            public ReqHeader head = new ReqHeader();
            public CommonBody body = new CommonBody();
        }

        public class CommonBody
        {
            public string ClientCode { get; set; }
        }

        public class LoginRequestV2Req
        {
            public ReqHeader head = new ReqHeader();
            public LoginRequestV2ReqBody body = new LoginRequestV2ReqBody();
        }

        public class LoginRequestV2ReqBody
        {
            public string ClientCode { get; set; }
            public string Password { get; set; }
            public string LocalIP { get; set; }
            public string PublicIP { get; set; }
            public string HDSerialNumber { get; set; }
            public string MACAddress { get; set; }
            public string MachineID { get; set; }
            public string VersionNo { get; set; }
            public string RequestNo { get; set; }
            public string My2PIN { get; set; }
            public string ConnectionType { get; set; }
        }

        public class LoginRequestMobileReq
        {
            public ReqHeader head = new ReqHeader();
            public LoginRequestMobileReqBody body = new LoginRequestMobileReqBody();
        }

        public class ReqHeader
        {
            public string appName { get; set; }
            public string appVer { get; set; }
            public string key { get; set; }
            public string osName { get; set; }
            public string requestCode { get; set; }
            public string userId { get; set; }
            public string password { get; set; }
        }

        public class LoginRequestMobileReqBody
        {
            public string Email_id { get; set; }
            public string LocalIP { get; set; }
            public string PublicIP { get; set; }
            public string ContactNumber { get; set; }
        }
    }
}

File: examples/vbnet/CommonCode.vb
Imports System.Net

Public Class CommonCode
    Public Function GetAppKeySettings(ByVal sKey As String) As String
        Return ConfigurationManager.AppSettings(sKey).ToString()
    End Function

    Public Function GetIPAddress() As String
        Dim strHostName As String = ""
        Dim strIPAddress = ""

        Try
            strHostName = Net.Dns.GetHostName()
            Dim ipaddress As IPAddress() = Net.Dns.GetHostAddresses(strHostName)

            For Each ip As IPAddress In ipaddress
                strIPAddress = ip.ToString()
            Next

        Catch ex As Exception
            strIPAddress = ""
        End Try

        Return strIPAddress
    End Function

    Public Class LoginRequestMobileRes
        Public head As ResHeader = New ResHeader()
        Public body As LoginRequestMobileResBody = New LoginRequestMobileResBody()
    End Class

    Public Class LoginRequestMobileResBody
        Public Property EmailId As String
        Public Property Message As String
        Public Property Status As Integer
        Public Property TTManagerId As String
        Public Property UserName As String
        Public Property ValidUpto As Date
        Public Property VendorName As String
    End Class

    Public Class ResHeader
        Public Property responseCode As String
        Public Property status As String
        Public Property statusDescription As String
    End Class

    Public Class BOCommonReq
        Public head As ReqHeader = New ReqHeader()
        Public body As BOCommonReqBody = New BOCommonReqBody()
    End Class

    Public Class BOCommonReqBody
        Public Property ClientCode As String
        Public Property FromDate As String
        Public Property ToDate As String
    End Class

    Public Class OrderStatusReq
        Public head As ReqHeader = New ReqHeader()
        Public body As OrderStatusReqBody = New OrderStatusReqBody()
    End Class

    Public Class OrderStatusReqBody
        Public Property ClientCode As String
        Public OrdStatusReqList As List(Of OrderStatusList) = New List(Of OrderStatusList)()
    End Class

    Public Class OrderStatusList
        Public Property Exch As Char
        Public Property ExchType As Char
        Public Property ScripCode As Long
        Public Property RemoteOrderID As String
    End Class

    Public Class TradeInformationReq
        Public head As ReqHeader = New ReqHeader()
        Public body As TradeInformationReqBody = New TradeInformationReqBody()
    End Class

    Public Class TradeInformationReqBody
        Public Property ClientCode As String
        Public TradeInformationList As List(Of TradeInformationList) = New List(Of TradeInformationList)()
    End Class

    Public Class TradeInformationList
        Public Property Exch As Char
        Public Property ExchType As Char
        Public Property ScripCode As Long
        Public Property ExchOrderID As String
    End Class

    Public Class PreOrdMarginCalReq
        Public head As ReqHeader = New ReqHeader()
        Public body As PreOrdMarginCalReqBody = New PreOrdMarginCalReqBody()
    End Class

    Public Class PreOrdMarginCalReqBody
        Public Property ClientCode As String
        Public Property OrderRequestorCode As String
        Public Property Exch As String
        Public Property ExchType As String
        Public Property ScripCode As Integer
        Public Property PlaceModifyCancel As String
        Public Property TransactionType As String
        Public Property AtMarket As String
        Public Property LimitRate As Double
        Public Property WithSL As String
        Public Property SLTriggerRate As Double
        Public Property IsSLTriggered As Char
        Public Property Volume As Long
        Public Property OldTradedQty As Long
        Public Property ProductType As Char
        Public Property ExchOrderId As String
        Public Property ClientIP As String
        Public Property AppSource As Integer
    End Class

    Public Class CommonReq
        Public head As ReqHeader = New ReqHeader()
        Public body As CommonBody = New CommonBody()
    End Class

    Public Class CommonBody
        Public Property ClientCode As String
    End Class

    Public Class LoginRequestV2Req
        Public head As ReqHeader = New ReqHeader()
        Public body As LoginRequestV2ReqBody = New LoginRequestV2ReqBody()
    End Class

    Public Class LoginRequestV2ReqBody
        Public Property ClientCode As String
        Public Property Password As String
        Public Property LocalIP As String
        Public Property PublicIP As String
        Public Property HDSerialNumber As String
        Public Property MACAddress As String
        Public Property MachineID As String
        Public Property VersionNo As String
        Public Property RequestNo As String
        Public Property My2PIN As String
        Public Property ConnectionType As String
    End Class

    Public Class LoginRequestMobileReq
        Public head As ReqHeader = New ReqHeader()
        Public body As LoginRequestMobileReqBody = New LoginRequestMobileReqBody()
    End Class

    Public Class ReqHeader
        Public Property appName As String
        Public Property appVer As String
        Public Property key As String
        Public Property osName As String
        Public Property requestCode As String
        Public Property userId As String
        Public Property password As String
    End Class

    Public Class LoginRequestMobileReqBody
        Public Property Email_id As String
        Public Property LocalIP As String
        Public Property PublicIP As String
        Public Property ContactNumber As String
    End Class


End Class
